src/Passes/Source/RuleTransformationPass/RulePass.hpp

```diff
@@ -0,0 +1,245 @@
+{
+namespace quantum
+{
+
+    /// This class applies a set of transformation rules to the IR to transform it into a new IR. The
+    /// rules are added using the RuleSet class which allows the developer to create one or more rules
+    /// on how to transform the IR.
+    ///
+    ///
+    /// The module executes the following steps:
+    ///
+    ///
+    ///
+    ///           ┌─────────────────┐
+    ///           │  Apply profile  │
+    ///           └─────────────────┘
+    ///                    │
+    ///                    │
+    ///                    │
+    ///                    │
+    ///                    │                ┌───────────────────────────────┐
+    ///                    │                │                               │
+    ///                    ├───────────────▶│   Copy and expand functions   │──┐
+    ///                    │     clone      │                               │  │
+    ///                    │   functions?   └───────────────────────────────┘  │
+    ///                    │                                │ delete dead      │
+    ///                    │                                │    code?         │
+    ///                    │                                ▼                  │
+    ///                    │                ┌───────────────────────────────┐  │
+    ///                    │                │                               │  │
+    ///                    ├───────────────▶│     Determine active code     │  │
+    ///                    │  delete dead   │                               │  │
+    ///                    │     code?      └───────────────────────────────┘  │
+    ///                    │                                │                  │
+    ///                    │                                │                  │
+    ///                    │                                ▼                  │
+    ///                    │                ┌───────────────────────────────┐  │  leave dead
+    ///                    │                │                               │  │    code?
+    ///                    │                │      Simplify phi nodes       │  │
+    ///                    │                │                               │  │
+    ///                    │                └───────────────────────────────┘  │
+    ///                    │                                │                  │
+    ///                    │                                │                  │
+    ///                    │                                ▼                  │
+    ///                    │                ┌───────────────────────────────┐  │
+    ///                    │                │                               │  │
+    ///                    │                │       Delete dead code        │  │
+    ///                    │                │                               │  │
+    ///                    │                └───────────────────────────────┘  │
```

Comment:
Consider minimizing:
```suggestion
    ///                    │                ┌───────────────────────────────┐  │
    ///                    │                │       Delete dead code        │  │
    ///                    │                └───────────────────────────────┘  │
```

Reply:
Fixed.